dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/advanced/DateRangeValidator.java

```diff
@@ -0,0 +1,114 @@
+package edu.harvard.iq.dataverse.search.advanced;
+
+import edu.harvard.iq.dataverse.common.BundleUtil;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.time.DateUtils;
+
+import javax.faces.application.FacesMessage;
+import javax.faces.component.UIComponent;
+import javax.faces.component.UIInput;
+import javax.faces.context.FacesContext;
+import javax.faces.validator.FacesValidator;
+import javax.faces.validator.Validator;
+import javax.faces.validator.ValidatorException;
+import java.text.ParseException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+
+@FacesValidator(value = "dateRangeValidator")
+public class DateRangeValidator implements Validator {
+
+    private static final String[] DATE_FORMATS = {"yyyy", "-yyyy", "yyyy-MM", "-yyyy-MM", "yyyy-MM-dd", "-yyyy-MM-dd"};
+    private static final String DATE_PATTERN = "^(\\-?)[0-9]{4}((\\-)([0-9]{2})){0,2}$";
+
+    // -------------------- LOGIC --------------------
+    @Override
+    public void validate(FacesContext context, UIComponent component, Object value) throws ValidatorException {
+        UIInput dateFromInput = (UIInput) component.getAttributes().get("dateFrom");
+        UIInput dateToInput = (UIInput) component.getAttributes().get("dateTo");
+
+        Object dateFromValue = dateFromInput.getSubmittedValue();
+        Object dateToValue = dateToInput.getSubmittedValue();
+
+        Optional<LocalDate> dateFrom = validateDate(context, dateFromInput, dateFromValue);
+        Optional<LocalDate> dateTo = validateDate(context, dateToInput, dateToValue);
+
+        if(dateFrom.isPresent() && dateTo.isPresent() && !isValidDateRange(dateFrom.get(), dateTo.get(), dateToValue.toString())) {
+            dateFromInput.setValid(false);
+            dateToInput.setValid(false);
+            FacesMessage message = new FacesMessage(FacesMessage.SEVERITY_ERROR, "", BundleUtil.getStringFromBundle("advanced.search.wrong.daterange.badRange"));
+            context.addMessage(dateFromInput.getClientId(context), message);
+            context.addMessage(dateToInput.getClientId(context), message);
+        }
+    }
+
+    // -------------------- PRIVATE ---------------------
+    private Optional<LocalDate> validateDate(FacesContext context, UIComponent comp, Object value) {
+        if(value == null || value.toString().isEmpty()) {
+            return Optional.empty();
+        }
+
+        try {
+            isDatePatternCorrect(value);
```

Comment:
Consider using `Either` or some similar construct instead of throwing an exception.